subprojects/plugin-development/src/integTest/groovy/org/gradle/plugin/devel/tasks/RuntimePluginValidationIntegrationTest.groovy

```diff
@@ -123,4 +122,200 @@ class RuntimePluginValidationIntegrationTest extends AbstractPluginValidationInt
+                }
+
+                @Nested
+                public Iterable<Options> getIterableOptions() {
+                    return Arrays.asList(new Options());
+                }
+
+                @Nested
+                public Iterable<Iterable<Options>> getDoubleIterableOptions() {
+                    return Arrays.asList(Arrays.asList(new Options()));
+                }
+
+                @Nested
+                public Map<String, Options> getMappedOptions() {
+                    return Collections.singletonMap("alma", new Options());
+                }
+
+                @Nested
+                public Iterable<Map<String, Iterable<Options>>> getIterableMappedOptions() {
+                    return Arrays.asList(Collections.singletonMap("alma", Arrays.asList(new Options())));
+                }
+
+                @Nested
+                public Provider<Options> getProvidedOptions() {
+                    return getProject().getObjects().property(Options.class).convention(new Options());
+                }
+
+                @Nested
+                public Iterable<NamedBean> getNamedIterable() {
+                    return Arrays.asList(new NamedBean());
+                }
+
+                @Nested
+                public AnnotatedList getAnnotatedList() {
+                    return new AnnotatedList();
+                }
+
+                public static class Options {
+                    @Input
+                    public String getGood() {
+                        return "good";
+                    }
+
+                    public String getNotAnnotated() {
+                        return null;
+                    }
+                }
+
+                public static class NamedBean implements Named {
+                    @Input
+                    public String getGood() {
+                        return "good";
+                    }
+
+                    public String getNotAnnotated() {
+                        return null;
+                    }
+
+                    @Internal
+                    public String getName() {
+                        return "tibor";
+                    }
+                }
+
+                // Does not validate the type parameter of extended collection
+                // because it has annotated properties
+                public static class AnnotatedList extends ArrayList<Options> {
+                    public AnnotatedList() {
+                        add(new Options());
+                    }
+
+                    @Input
+                    public String getGood() {
+                        return "good";
+                    }
+                }
+
+                @TaskAction
+                public void doStuff() { }
+            }
+        """
+
+        expect:
+        executer.withArgument("-Dorg.gradle.internal.max.validation.errors=10")
+        assertValidationFailsWith([
+                // TODO: not sure why these were not detected
```

Comment:
We should resolve this TODO before merging.

Reply:
This should be fixed now.